spec/lib/ember-cli/path_set_spec.rb

```diff
@@ -0,0 +1,265 @@
+require "fileutils"
+require "ember-cli/path_set"
+
+describe EmberCli::PathSet do
+  describe "#root" do
+    it "depends on the app name" do
+      app = build_app(name: "foo")
+
+      path_set = build_path_set(app: app)
+
+      expect(path_set.root).to eq rails_root.join("foo")
+    end
+
+    it "can be overridden" do
+      app = build_app(name: "foo", options: { path: "not-foo" })
+
+      path_set = build_path_set(app: app)
+
+      expect(path_set.root).to eq rails_root.join("not-foo")
+    end
+  end
+
+  describe "#tmp" do
+    it "is a child of #root" do
+      app = build_app
+
+      path_set = build_path_set(app: app)
+
+      expect(path_set.tmp).to exist
+      expect(path_set.tmp).to eq rails_root.join(app.name, "tmp")
+    end
+  end
+
+  describe "#tmp" do
+    it "depends on the environment" do
+      app = build_app(name: "foo")
+
+      path_set = build_path_set(app: app, environment: "bar")
+
+      expect(path_set.log).to eq rails_root.join("log", "ember-foo.bar.log")
+    end
+  end
+
+  describe "#dist" do
+    it "depends on the app name" do
+      app = build_app(name: "foo")
+
+      path_set = build_path_set(app: app)
+
+      expect(path_set.dist).to exist
+      expect(path_set.dist).to eq ember_cli_root.join("apps", "foo")
+    end
+  end
+
+  describe "#assets" do
+    it "is a child of #ember_cli_root" do
+      path_set = build_path_set
+
+      expect(path_set.assets).to exist
+      expect(path_set.assets).to eq ember_cli_root.join("assets")
+    end
+  end
+
+  describe "#applications" do
+    it "is a child of #rails_root" do
+      path_set = build_path_set
+
+      expect(path_set.applications).to exist
+      expect(path_set.applications).to eq rails_root.join("public", "_apps")
+    end
+  end
+
+  describe "#gemfile" do
+    it "is a child of #root" do
+      app = build_app(name: "foo")
+
+      path_set = build_path_set(app: app)
+
+      expect(path_set.gemfile).to eq rails_root.join("foo", "Gemfile")
+    end
+  end
+
+  describe "#tests" do
+    it "is a child of #dist" do
+      path_set = build_path_set
+
+      expect(path_set.tests).to eq path_set.dist.join("tests")
+    end
+  end
+
+  describe "#node_modules" do
+    it "is a child of #root" do
+      path_set = build_path_set
+
+      expect(path_set.node_modules).to eq path_set.root.join("node_modules")
+    end
+  end
+
+  describe "#package_json_file" do
+    it "is a child of #root" do
+      path_set = build_path_set
+
+      expect(path_set.package_json_file)
+        .to eq path_set.root.join("package.json")
+    end
+  end
+
+  describe "#addon_package_json_file" do
+    it "is a child of #root" do
+      path_set = build_path_set
+
+      expect(path_set.addon_package_json_file)
+        .to eq path_set.node_modules.join("ember-cli-rails-addon", "package.json")
```

Comment:
Line is too long. [82/80]<br>Place the . on the previous line, together with the method call receiver.